### ISSUE TITLE
Using i3blocks in terminal

### ISSUE THREAD

**Issue by pickfire** [vivien/i3blocks#134]
Use `i3blocks` just like `i3status` in terminal with the color output.

**Comment by kb100** [comment]
Hi @pickfire, could you elaborate on the problem you are having? It's not clear what the issue is.

**Comment by Zopieux** [comment]
Works fine here.

```
$ i3blocks
{"version":1,"click_events":true}
[[]
,[{"full_text":""}]
,[{"full_text":""},{"":"","full_text":"[] Paul Kalkbrenner - Castenets (Special Berlin Calling edit)","color":"#aaaaaa","name":"MPD","separator_block_width":15,"markup":"none"}]
...
```

**Comment by pickfire** [comment]
No, I mean it will just show the colored output.

![2015-09-03-214034_564x340_scrot](https://cloud.githubusercontent.com/assets/4687791/9659706/76844002-5284-11e5-87a0-09c81194d6dd.png)

**Comment by Zopieux** [comment]
Ho okay, you want a preview feature. I guess it would be low priority but it can be a nice feature to have.

**Comment by pickfire** [comment]
Yeah, low priority, just for debugging purpose or maybe it could also work on other window manager such as dwm.

**Comment by vivien** [comment]
Indeed that'd be low priority, but I'm not against it. I'm actually experimenting dynamic block properties that implies a bit of refactoring. This might be introduced at the same time.

**Comment by pickfire** [comment]
> I'm actually experimenting dynamic block properties that implies a bit of refactoring.

What do you mean by that?

**Comment by vivien** [comment]
Hi @pickfire 

Today i3bar and i3blocks properties are hardcoded in `block.h`. You can set known properties to blocks, but it'll fail on unknown ones (e.g. `foo=bar` is invalid).

This was a design choice, which makes blocks of a known size. There is no runtime allocation needed, resetting a block with its default values is just a matter of `memcpy`. This is fast and very convenient.

However, some concerns have been raised about that. When a new i3 version with a new i3bar property is released, I have to release a new i3blocks version supporting it. Also, @Airblader's i3-gap fork of i3bar is a good example for additional properties, which requires a small patch to i3blocks to work.

Having dynamic properties would also allow to support custom properties (i.e. user-prefixed by `_`), which might be very convenient to pass extra arguments through environment variables to the block command.

So my feeling right now is that i3blocks must go into the direction where it is not aware of the processing program behind it (here i3bar). It is simply a flexible scheduler for user commands that must take key-value blocks, process them, and output them in the wanted format (that's where a "console" output format beside the current "json" might be introduced).

I have a bit of work in progress for that. If the implied runtime allocation doesn't introduce a substantial performance issue, I'd gladly go that way.

**Comment by scarlehoff** [comment]
What is the status of this?
If nobody is working on this I might give it a go (I want to use i3blocks with dwm).

**Comment by kb100** [comment]
@scarlehoff Terminal output has not been pursued. Please feel free to work on it if you so desire.

**Comment by prithugoswami** [comment]
Great! 

3887a35 adds the terminal output but It also prints out terminal escape sequences. Perhaps there can a plain text output as well which can be used with other bars?
